sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -2,8 +2,9 @@
 // Licensed under the MIT License.
 
 using System;
+using System.ClientModel;
```

Comment:
There seems to be a lot of complex networking code here. I was expecting that all this code would be in client model and then Azure.Core would have a very simple call forwarding adapter. If we cannot get to that point, is there a good reason for all this inheritance or we should have two separate pipelines?

Reply:
Are you referring to lines 126-224?  This is for cookie, proxy, and cert handling that I had assumed were specific to Azure clients.  If we feel they are general enough to move into ClientModel, we can do that.  The cookie we have is named "Azure.Core.Pipeline.HttpClientTransport.EnableCookies" so we would probably want a different name for that one.

Reply:
It's not any single part of the code. It's the overall feeling that the resulting Azure.Core is now even harder to understand than it was before. Possibly I am wrong about this, but I want to make sure we don't end up with a mess in Azure.Core.